### PR TITLE
[#126384395] Recurring schedules are persisted

### PR DESCRIPTION
**Pull request by kanekoh**
Added a new table for recurring schedule, new entity and DAO class and modified other classes to support recurring schedule persistence.

### CODE REVIEW

File: scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java [hunk 1]
```diff
@@ -141,10 +159,19 @@ public void validateSchedules(String appId, ApplicationScalingSchedules applicat
...
+			// Validate specific date schedules.
+			if (specificDateSchedules != null) {
+				validateSpecificDateSchedules(specificDateSchedules, isValidTimeZone);
+			}
 
-			validateSpecificDateSchedules(applicationScalingSchedules.getSpecific_date(), isValidTimeZone);
+			List<RecurringScheduleEntity> recurringSchedules = applicationScalingSchedules.getRecurring_schedule();
+			// Validate recurring schedules.
+			if (recurringSchedules != null) {
+				//TODO validate recurring schedule.
```
Comment (fraenkel) [inline scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java]: I don't see any story for validation therefore this needs to be done.

File: scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java [hunk 2]
```diff
@@ -408,13 +429,28 @@ private SpecificDateScheduleDateTime validateStartEndDateTime(String scheduleBei
...
+		}
+
+		List<RecurringScheduleEntity> recurringScheduls = applicationScalingSchedules.getRecurring_schedule();
+		if (recurringScheduls != null) {
+			for (RecurringScheduleEntity recurringScheduleEntity : recurringScheduls) {
+				// Persist the schedule in database
+				RecurringScheduleEntity savedScheduleEntity = saveNewRecurringSchedule(recurringScheduleEntity);
+
+				// Ask ScalingJobManager to create scaling job
+				if (savedScheduleEntity != null) {
```
Comment (fraenkel) [inline scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java]: What is supposed to be here?

File: scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java (hunk 1 above)
Comment (kanekoh) [inline scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java]: Will create a new story for validation of recurring schedule.

File: scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java (hunk 2 above)
Comment (kanekoh) [inline scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java]: Be supposed to implement to create cron job with Quartz on the story 'When recurring rules are created, associated jobs are created'.
Will remove the 'if' statement.

**Comment by fraenkel** [comment]
Merged

**Pull request closed without merging**